snippets/partition_options.mdx

```diff
@@ -0,0 +1,25 @@
+- `extract_images`: A boolean that determines whether to extract images from the document. Default: `False`.
+- `ocr_images`: A boolean that, when `True`, causes DocParse to use OCR to attempt to generate a text representation of detected images. When `False`, images do not contain a text_representation. Default is `False`.
+- `selected_pages`: A list specifying individual pages (1-indexed) and page ranges from the document to partition. Single pages are specified as integers and ranges are specified as lists with two integer entries in ascending order. A valid example value for selected_pages is `[1, 10, [15, 20]]` which would include pages 1, 10, 15, 16, 17 ..., 20.  `selected_pages` is `None` by default, which results in all pages of the document being parsed.
+- `chunking_options`: A dictionary of options for specifying chunking behavior. Chunking is only performed when this option is present, and default options are chosen when `chunking_options` is specified as `{}`.
+    - `strategy`: A string specifying the strategy to use to combine and split chunks. Valid values are `context_rich`, `mixed_multi_column`, and `maximize_within_limit`. The default and recommended chunker is `context_rich` as `{'strategy': 'context_rich'}`.
+        - Behavior of `context_rich` chunker: The goal of this strategy is to add context to evenly-sized chunks. This is most useful for retrieval based GenAI applications. context_rich chunking combines adjacent `Section-header` and `Title` elements into a new `Section-header` element. Merges elements into a chunk with its most recent `Section-header`. If the chunk would contain too many tokens, then it starts a new chunk copying the Section-header to the start of this new chunk and continues. Merges elements on different pages, unless `merge_across_pages` is set to `False`.
+
+        - Behavior of `mixed_multi_column` chunker: `mixed_multi_column` performs well on journal articles and documents layouts that contain one or two columns. For each page, it detects whether the elements are present in two columns and if so, it orders them properly with the left column appearing in the order first. Then, it drops elements that are only one token in size. It also drops elements in the top and bottom 5% of the page. Elements returned do not have a `type`. The `marked` merger always breaks across pages.
+
+        - Behavior of `maximize_within_limit` chunker: The goal of the `maximize_within_limit` chunker is to make the chunks as large as possible. Merges elements into the last most recently merged set of elements unless doing so would make its token count exceed `max_tokens`. In that case, it would keep the new element separate and start merging subsequent elements into that one, following the same rule. Merges elements on different pages, unless `merge_across_pages` is set to `False`.
+    - `max_tokens`: An integer specifying the cutoff for splitting chunks that are too large. Default value is 512.
```

Comment:
Is there a limit to how large this can get? Could someone make this large enough that it fails the request because of memory limits? Worse, could they make this large enough to cause the service to crash?

Reply:
Discussed.